code/obj/machinery/plantpot.dm

```diff
@@ -741,30 +717,34 @@
 			pingsignal.transmission_method = TRANSMISSION_RADIO
 
 			var/datum/radio_frequency/frequency = radio_controller.return_frequency("[report_freq]")
-			if (!frequency) return
+			if(!frequency) return
 			SPAWN_DBG(0.5 SECONDS) //Send a reply for those curious jerks
 				frequency.post_signal(src, pingsignal)
 
 		return //Just toss out the rest of the signal then I guess
 
 	// Procs specific to the plantpot start here.
 
-	proc/update_water()
+	proc/update_water_icon()
+		var/datum/color/average
 		src.water_sprite = image('icons/obj/hydroponics/hydroponics.dmi',"wat-[src.total_volume]")
 		src.water_display = image('icons/obj/hydroponics/hydroponics.dmi',"ind-wat-[src.water_level]")
 
-		if (src.reagents.total_volume)
-			var/datum/color/average = src.reagents.get_average_color()
-			src.water_sprite.color = average.to_rgba()
+		if(src.reagents.total_volume)
+			average = src.reagents.get_average_color()
+			if(average.to_rgba() != src.water_sprite.color)
+				src.water_sprite.color = average.to_rgba()
+			else
+				average = null
 
-		UpdateOverlays(src.water_display, "water")
-		UpdateOverlays(src.water_sprite, "water_fluid")
+		if(average || !current) //only update water level if there is no plant in the tray or reagent contents change color (cheap fix to a layer stacking issue)
```

Comment:
I feel like you should update the water level even if there's a plant in the tray. What's the layer stacking issue?

Reply:
a plant growing updates the plant overlay, which causes the plant to appear above the water sprite.

Then if the water level changes, the water updates and is pasted over the plant. This is an issue that exists in current plant pots, so this check makes it so only the water meter updates, not the water level sprite. I *could* explicitly set layers to the overlay images (which im considering), but it causes some weird things to happen when you drag plant pots over other items, though that case is slim enough that more people would notice water not updating visually until the plant is gone.